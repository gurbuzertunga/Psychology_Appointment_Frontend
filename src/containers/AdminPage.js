import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import ConsultancyForm from '../components/consultancyForm';

const AdminPage = props => {
  const { authToken } = props;
  if (!authToken) {
    return <Redirect to="/" />;
  }
  return (
    <div className="flex flex-col justify-evenly items-center mt-16">
      <h1 className="text-white lg:text-2xl text-4xl font-bold mb-4">Admin Page</h1>
      <ConsultancyForm />
    </div>
  );
};

AdminPage.propTypes = {
  authToken: PropTypes.string,
};
AdminPage.defaultProps = {
  authToken: '',
};
const mapStateToProps = state => ({
  authToken: state.authenticationReducer.authToken,
});

export default connect(mapStateToProps)(AdminPage);
