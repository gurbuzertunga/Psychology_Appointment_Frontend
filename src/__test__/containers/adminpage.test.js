import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import AdminPage from '../../containers/AdminPage';

configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <AdminPage />
  </Provider>,
);

describe('AdminPage Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the Consultancies List Component', () => {
    const page = component.find('.adminPage');
    expect(page).toMatchSnapshot();
  });
});
