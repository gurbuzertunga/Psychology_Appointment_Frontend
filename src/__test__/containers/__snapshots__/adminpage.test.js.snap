// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPage Component renders the Consultancies List Component 1`] = `ShallowWrapper {}`;
